craft_application/services/remotebuild.py

```diff
@@ -62,13 +62,43 @@ def __init__(self, app: AppMetadata, services: ServiceFactory) -> None:
         self._repository: launchpad.models.GitRepository = None  # type: ignore[assignment]
         self._recipe: launchpad.models.recipe.BaseRecipe = None  # type: ignore[assignment]
         self._builds: Collection[launchpad.models.Build] = []
+        self._project_name: str | None = None
 
     def setup(self) -> None:
         """Set up the remote builder."""
         self.lp = self._get_lp_client()
 
     # region Public API
     # Commands will call a subset of these methods, generally in order.
+    def set_project(self, name: str) -> None:
+        """Set the name of the project to use."""
+        if self._is_setup:
+            raise RuntimeError(
+                "Project name may only be set before starting or resuming builds."
+            )
+        self._project_name = name
```

Comment:
possibly pedantic, but maybe do this setting after the try/except block, to avoid having an 'invalid' object that has a `_project_name` but no matching `_lp_project`

Reply:
Good point! Right now it doesn't matter since this should result in an app exit, but we can't necessarily trust an app to do that. Updated.